internal/resource/utils/tenant.go

```diff
@@ -24,59 +24,97 @@ import (
 
 	"github.com/oceanbase/ob-operator/api/v1alpha1"
 	oceanbaseconst "github.com/oceanbase/ob-operator/internal/const/oceanbase"
+	"github.com/oceanbase/ob-operator/pkg/oceanbase-sdk/operation"
 )
 
 // GetTenantRestoreSource gets restore source from tenant CR. If tenantCR is in form of ns/name, the parameter ns is ignored.
 func GetTenantRestoreSource(ctx context.Context, clt client.Client, logger *logr.Logger, ns, tenantCR string) (string, error) {
+	primary, con, err := getTenantAndClusterCon(ctx, clt, logger, ns, tenantCR)
+	if err != nil {
+		return "", err
+	}
+	// Get ip_list from primary tenant
+	aps, err := con.ListTenantAccessPoints(primary.Spec.TenantName)
+	if err != nil {
+		return "", err
+	}
+	ipList := make([]string, 0)
+	for _, ap := range aps {
+		ipList = append(ipList, fmt.Sprintf("%s:%d", ap.SvrIP, ap.SqlPort))
+	}
+	standbyRoPwd, err := ReadPassword(clt, ns, primary.Status.Credentials.StandbyRO)
+	if err != nil {
+		logger.Error(err, "Failed to read standby ro password")
+		return "", err
+	}
+	// Set restore source
+	restoreSource := fmt.Sprintf("SERVICE=%s USER=%s@%s PASSWORD=%s", strings.Join(ipList, ";"), oceanbaseconst.StandbyROUser, primary.Spec.TenantName, standbyRoPwd)
+
+	return restoreSource, nil
+}
+
+// CheckTenantLSIntegrity checks LS integrity of tenant CR. If tenantCR is in form of ns/name, the parameter ns is ignored.
+func CheckTenantLSIntegrity(ctx context.Context, clt client.Client, logger *logr.Logger, ns, tenantCR string) error {
+	primary, con, err := getTenantAndClusterCon(ctx, clt, logger, ns, tenantCR)
+	if err != nil {
+		return err
+	}
+	// Check LS integrity
+	lsDeletion, err := con.ListLSDeletion(int64(primary.Status.TenantRecordInfo.TenantID))
+	if err != nil {
+		return err
+	}
+	if len(lsDeletion) > 0 {
+		return errors.New("LS deletion set is not empty, log is of not integrity")
+	}
+	logStats, err := con.ListLogStats(int64(primary.Status.TenantRecordInfo.TenantID))
+	if err != nil {
+		return err
+	}
+	if len(logStats) == 0 {
+		return errors.New("Log stats is empty, out of expectation")
+	}
+	for _, ls := range logStats {
+		if ls.BeginLSN != 0 {
+			return errors.New("Log stats begin SCN is not 0, log is of not integrity")
+		}
+	}
+
+	return nil
+}
+
+func getTenantAndClusterCon(ctx context.Context, clt client.Client, logger *logr.Logger, ns, tenantCR string) (*v1alpha1.OBTenant, *operation.OceanbaseOperationManager, error) {
```

Comment:
split this into two function, get tenant and get sys connection
besides, the tenant may not be primary, so the parameter 'primary' can be changed to a more general name